src/pages/LogOutPreviousUserPage.tsx

```diff
@@ -22,10 +28,12 @@ type LogOutPreviousUserPageProps = LogOutPreviousUserPageOnyxProps & StackScreen
 // out if the transition is for another user.
 //
 // This component should not do any other navigation as that handled in App.setUpPoliciesAndNavigate
-function LogOutPreviousUserPage({session, route}: LogOutPreviousUserPageProps) {
+function LogOutPreviousUserPage({session, route, account}: LogOutPreviousUserPageProps) {
+    const initUrl = useContext(InitialUrlContext);
     useEffect(() => {
-        Linking.getInitialURL().then((transitionURL) => {
+        Linking.getInitialURL().then((url) => {
             const sessionEmail = session?.email;
+            const transitionURL = NativeModules.HybridAppModule ? CONST.DEEPLINK_BASE_URL + initUrl : url;
             const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(transitionURL ?? undefined, sessionEmail);
 
             if (isLoggingInAsNewUser) {
```

Comment:
quick question, in other places we `return` when we call the `Session.signInWithShortLivedAuthToken` and `signOutAndRedirectToSignIn` methods. Do we need to do that here?